### PR TITLE
Best Practices

### PR DESCRIPTION
**Pull request by Shivam-TI**
The changes to be sorted out before reviewing:

- [x] Update README
- [x] Follow standard naming conventions for classes and functions
- [x] Add .gitignore
- [x] Add .env
- [x] Update requirements.txt
- [x] Remove hardcoded status codes
- [x] Move hardcoded constants
- [x] Remove unused imports and import all
- [x] Add static typing
- [x] Add logging
- [x] Add Docstrings

### CODE REVIEW

**Comment by Shivam-TI** [comment]
Hi @kushal-ti, this PR is ready to be reviewed.

File: restapi/utils.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    return result
+
+
+def reader(url, timeout):
+    """Read file through HTTP"""
+    with urllib.request.urlopen(url, timeout=timeout) as conn:
+        return conn.read()
+
+
+def multi_threaded_reader(urls, num_threads) -> list:
```
Comment (kushal-ti) [inline restapi/utils.py:85]: method can be optimized by using proper concurrency primitives

**Review by kushal-ti — CHANGES_REQUESTED** [review]
Really good. just have one change. Also, static typing could be a bit more extensive, but this is fine too

File: restapi/utils.py (hunk 1 above)
Comment (Shivam-TI) [reply restapi/utils.py:86]: I have now changed the method to use ThreadPoolReader.

**Pull request closed without merging**